sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -129,11 +143,28 @@ export const imdsMsi: MSI = {
       `Using the IMDS endpoint coming form the environment variable MSI_ENDPOINT=${process.env.MSI_ENDPOINT}, and using the cloud shell to proceed with the authentication.`
     );
 
-    return msiGenericGetToken(
-      identityClient,
-      prepareRequestOptions(resource, clientId),
-      expiresInParser,
-      getTokenOptions
+    for (let retries = 0; retries < imdsMsiRetryConfig.maxRetries; retries++) {
+      try {
+        return await msiGenericGetToken(
+          identityClient,
+          prepareRequestOptions(resource, clientId),
+          expiresInParser,
+          getTokenOptions
+        );
+      } catch (error) {
+        if (error.statusCode === 404) {
+          const nextDelay = imdsMsiRetryConfig.nextDelayInMs;
+          imdsMsiRetryConfig.nextDelayInMs *= imdsMsiRetryConfig.intervalIncrement;
+          await delay(nextDelay);
```

Comment:
since you do the multiplication before the first delay, this ends up delaying for 60s on the first go around, which I don't believe if what you intended

Reply:
Thank you!

Reply:
No no, this is 30s on the first go, since I capture nextDelay (starts at 30s) before doing the multiplication. nextDelayInMs gets stored as 60s for the next round

Reply:
I'll make it cleaner:

```ts
          await delay(nextDelayInMs);
          nextDelayInMs *= imdsMsiRetryConfig.intervalIncrement;
```

Reply:
oh derp, you're right, because of the above issue